### PR TITLE
Add FOSSA workflow 

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

# Description

This PR adds a CI step to ensure that dependencies with bad licenses are not introduced, along with a README badge.

## Issue reference

Please reference the issue this PR is related to: https://github.com/dapr/dapr/issues/4158

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* (NA) Code compiles correctly
* (NA) Created/updated tests
* (NA) Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: .github/workflows/fossa.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# limitations under the License.
+#
+
+name: fossa
+on:
+  push:
+    branches:
+      - master
+      - release-*
+      - feature/*
```
Comment (daixiang0) [inline .github/workflows/fossa.yml:20]: We do not have those, and deny those push.

File: .github/workflows/fossa.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    tags:
+      - v*
+  pull_request:
+    branches:
+      - master
+      - release-*
+      - feature/*
```
Comment (daixiang0) [inline .github/workflows/fossa.yml:27]: Same as above.

File: .github/workflows/fossa.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  workflow_dispatch: {}
+jobs:
+  fossa-scan:
+    if: github.repository_owner == 'dapr' # FOSSA is not intended to run on forks.
+    runs-on: ubuntu-latest
+    env:
+      FOSSA_API_KEY: b88e1f4287c3108c8751bf106fb46db6 # This is a push-only token that is safe to be exposed.
```
Comment (daixiang0) [inline .github/workflows/fossa.yml:34]: Seems it is yours?

Comment (shubham1172) [reply .github/workflows/fossa.yml:32]: @daixiang0 it's the FOSSA token from CNCF. Same that we use for other repositories.

Comment (daixiang0) [reply .github/workflows/fossa.yml:32]: I refer to making it hide and getting it from GH ENV to reduce potential risks。

Comment (shubham1172) [reply .github/workflows/fossa.yml:32]: FOSSA's push-only token is safe to be used in CI directly 
https://github.com/marketplace/actions/fossa-action#push-only-api-token
https://docs.fossa.com/docs/api-reference#push-only-api-token 

Similar example with Keda: https://github.com/kedacore/keda/blob/main/.github/workflows/fossa.yml

**Review by daixiang0 — APPROVED** [review]

**Review by mukundansundar — APPROVED** [review]
lgtm

**Comment by daixiang0** [comment]
@shubham1172 thanks for your contribution!

**Pull request merged**